crates/solver/src/driver.rs

```diff
@@ -418,10 +426,14 @@ impl Driver {
         id
     }
 
-    async fn send_solver_competition(&self, body: SolverCompetition) {
-        match self.api.send_solver_competition(&body).await {
+    async fn send_solver_competition(&self, body: &SolverCompetition) {
+        // For example shadow solver shouldn't store competition info.
+        if !self.store_competition_info {
```

Comment:
very nit-picky, but can we instead have this be:
```rust
self.api.is_authenticated()
```

That is, we let the `API` decide if its authenticated or not. Alternatively, we can have:
```rust
fn send_solver_competition(...) -> Result<(), SolverCompetitionError> {
    ...
}

enum SolverCompetitionError {
    NotAuthorized,
    Other(...),
}
```

Reply:
Don't feel strongly about either of these though.